ttlcache@3.4.0: Tests fail with Go 1.25.9

Hi,

After the switch default Go for packages in Guix (1.24.13 -> 1.25.9) quic-go package started failing during check phase:

```
--- FAIL: Test_Cache_Get (0.00s)
    --- FAIL: Test_Cache_Get/Get_when_item_is_found (0.00s)
    --- FAIL: Test_Cache_Get/Get_with_call_loader_that_returns_nil_value_when_item_is_not_found (0.00s)
=== CONT  Test_Cache_Get/Get_when_TTL_extension_is_disabled_and_item_is_found
--- FAIL: Test_Cache_Get (0.00s)
    --- FAIL: Test_Cache_Get/Get_with_default_loader_that_returns_nil_value_when_item_is_not_found (0.00s)
=== CONT  Test_Cache_Get/Get_without_loader_when_item_is_not_found
--- FAIL: Test_Cache_Get (0.00s)
    --- FAIL: Test_Cache_Get/Get_without_loader_when_item_is_not_found (0.00s)
panic: testing: AllocsPerRun called during parallel test [recovered, repanicked]

goroutine 57 [running]:
testing.tRunner.func1.2({0x62cc80, 0x6e6e60})
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1872 +0x237
testing.tRunner.func1()
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1875 +0x35b
panic({0x62cc80?, 0x6e6e60?})
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/runtime/panic.go:783 +0x132
testing.AllocsPerRun(0x1?, 0xc000337d80?)
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/allocs.go:22 +0x1a5
github.com/jellydator/ttlcache/v3.allocsPerSingleRun(0x0?)
        /tmp/guix-build-go-github-com-jellydator-ttlcache-v3-3.4.0.drv-0/src/github.com/jellydator/ttlcache/v3/cache_test.go:1549 +0x3d
github.com/jellydator/ttlcache/v3.Test_Cache_Get.func7(0xc0002aafc0)
        /tmp/guix-build-go-github-com-jellydator-ttlcache-v3-3.4.0.drv-0/src/github.com/jellydator/ttlcache/v3/cache_test.go:672 +0x1f8
testing.tRunner(0xc0002aafc0, 0xc000280690)
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1934 +0xea
created by testing.(*T).Run in goroutine 52
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1997 +0x465
panic: testing: AllocsPerRun called during parallel test [recovered, repanicked]

goroutine 55 [running]:
testing.tRunner.func1.2({0x62cc80, 0x6e6e60})
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1872 +0x237
testing.tRunner.func1()
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1875 +0x35b
panic({0x62cc80?, 0x6e6e60?})
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/runtime/panic.go:783 +0x132
testing.AllocsPerRun(0x1?, 0xc000419d80?)
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/allocs.go:22 +0x1a5
github.com/jellydator/ttlcache/v3.allocsPerSingleRun(0x0?)
        /tmp/guix-build-go-github-com-jellydator-ttlcache-v3-3.4.0.drv-0/src/github.com/jellydator/ttlcache/v3/cache_test.go:1549 +0x3d
github.com/jellydator/ttlcache/v3.Test_Cache_Get.func7(0xc0002aa8c0)
        /tmp/guix-build-go-github-com-jellydator-ttlcache-v3-3.4.0.drv-0/src/github.com/jellydator/ttlcache/v3/cache_test.go:672 +0x1f8
testing.tRunner(0xc0002aa8c0, 0xc0002805f0)
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1934 +0xea
created by testing.(*T).Run in goroutine 52
        /gnu/store/0jbvyiwkmhriv1aawj7lcsb3lmsa22my-go-1.25.9/lib/go/src/testing/testing.go:1997 +0x465
FAIL    github.com/jellydator/ttlcache/v3       0.163s
FAIL
```
Inputs:
- go-github-com-jellydator-ttlcache-v3@3.4.0
  - go-github-com-stretchr-testify@1.11.1
  - go-go-uber-org-goleak@1.3.0
  - go-golang-org-x-sync@0.19.0

Full build log:

[503b59y3qvbkvxsy8inp5gj2x4y1r4-go-github-com-jellydator-ttlcache-v3-3.4.0.drv.gz](https://github.com/user-attachments/files/27285964/503b59y3qvbkvxsy8inp5gj2x4y1r4-go-github-com-jellydator-ttlcache-v3-3.4.0.drv.gz)

--- a/cache_test.go
+++ b/cache_test.go
@@ -4,6 +4,7 @@ import (
 	"container/list"
 	"context"
 	"fmt"
+	"runtime"
 	"sync"
 	"testing"
 	"time"
@@ -1538,15 +1539,27 @@ func addCacheItems(c *Cache[string, string], itemFn func(index int, key string) 
 }
 
 func allocsPerSingleRun(f func()) int {
-	// `testing.AllocsPerRun` "warms up" the function for a single run before
-	// measuring allocations, so we need to do nothing on the first run.
-	var firstRun bool
+	// Like testing.AllocsPerRun but safe to call while parallel tests are paused.
+	defer runtime.GOMAXPROCS(runtime.GOMAXPROCS(1))
 
-	return int(testing.AllocsPerRun(1, func() {
-		if !firstRun {
-			firstRun = true
-			return
-		}
-		f()
+	var memstats runtime.MemStats
+	runtime.ReadMemStats(&memstats)
+	mallocs := memstats.Mallocs
+
+	f()
+
+	runtime.ReadMemStats(&memstats)
+	return int(memstats.Mallocs - mallocs)
+}
+
+var allocSink *int
+
+func Test_allocsPerSingleRun_withPausedParallelTest(t *testing.T) {
+	t.Run("paused parallel sibling", func(t *testing.T) {
+		t.Parallel()
+	})
+
+	assert.Equal(t, 1, allocsPerSingleRun(func() {
+		allocSink = new(int)
 	}))
 }

--- a/cache_test.go
+++ b/cache_test.go
@@ -1552,14 +1552,14 @@ func allocsPerSingleRun(f func()) int {
 	return int(memstats.Mallocs - mallocs)
 }
 
-var allocSink *int
-
 func Test_allocsPerSingleRun_withPausedParallelTest(t *testing.T) {
 	t.Run("paused parallel sibling", func(t *testing.T) {
 		t.Parallel()
 	})
 
+	var allocSink *int
 	assert.Equal(t, 1, allocsPerSingleRun(func() {
 		allocSink = new(int)
 	}))
+	runtime.KeepAlive(allocSink)
 }

